adr/0002-reactive-rename.adoc

```diff
@@ -0,0 +1,70 @@
+= Rename Reactive extensions
+
+* Status: proposed
+* Date: 2023-10-17
+
+== Context and Problem Statement
+
+RESTEasy Reactive is the preferred REST stack in Quarkus, however in the almost 3 years since its release, we have seen
+folks of all experience misunderstand its capabilities and being led to believe that it forces reactive programming on them.
+
+With the advent of Virtual Threads in Java 19+, the problem becomes larger since in the minds of users, Reactive and Virtual Threads
+are two different worlds.
+
+The same problem applies to other extensions as well:
```

Comment:
The render doesn't show bullet points properly, I think empty line after this one should fix it.

https://github.com/quarkusio/quarkus/blob/826a25db947b39bb10d12cdad60f443a1c182a95/adr/0002-reactive-rename.adoc shows
<img width="991" alt="Screenshot 2023-10-20 at 9 13 46" src="https://github.com/quarkusio/quarkus/assets/925259/34f2d6b1-8c1c-4542-9aa5-ce935afca077">

Reply:
Good point! Fixed